index.bs

```diff
@@ -795,6 +743,562 @@ the following list:
+     written to.
+  1. [=Resolve=] |stream|'s [=closed promise=] with a [=new=] {{WebSocketCloseInfo}} constructed
+     with |code| and |reason|.
+ 1. Otherwise,
+  1. [=ReadableStream/Error=] |stream|'s [=WebSocketStream/readable stream=] with a [=new=]
+     {{WebSocketError}} constructed with |code| and |reason|.
+  1. [=WritableStream/Error=] |stream|'s [=WebSocketStream/writable stream=] with a [=new=]
+     {{WebSocketError}} constructed with |code| and |reason|.
+  1. [=Reject=] |stream|'s [=closed promise=] with a [=new=] {{WebSocketError}} constructed with
+     |code| and |reason|.
+</div>
+
+<hr>
+
+The [=task source=] for all [=tasks=] <a lt="queue a task">queued</a> in this section is the
+[=WebSocket task source=].
+
+<hr>
+
+
+## Stream operations ## {#stream-operations}
+
+<div algorithm="pull algorithm">
+
+To <dfn>pull bytes</dfn> from a {{WebSocketStream}} |stream|, if |stream| is currently applying
+backpressure, release backpressure.
+
+ Note: If any messages are queued, one will be handled immediately as a result.
+</div>
+
+<hr>
+
+<div algorithm="cancel algorithm">
+
+To <dfn>cancel</dfn> a {{WebSocketStream}} |stream| given |reason|, [=close using reason=] giving
+|stream| and |reason|.
+</div>
+
+<hr>
+
+<div algorithm="write algorithm">
+
+To <dfn>write</dfn> |chunk| to a {{WebSocketStream}} |stream|, the user agent must run these steps:
+ 1. Let |promise| be a new promise.
+ 1. If |chunk| is a {{BufferSource}},
+  1. Let |data| be a [=get a copy of the buffer source|copy of the bytes=].
+  1. Let |opcode| be a binary frame opcode.
+ 1. Otherwise,
+  1. Let |string| be the result of [=Converted to an IDL value|converting=] |chunk| to an IDL
+     {{USVString}}. If this throws an exception, return [=a promise rejected with=] the exception.
+  1. Let |data| be the result of [=UTF-8 encode|encoding=] |string|.
+  1. Let |opcode| be a text frame opcode.
+ 1. In parallel,
+  1. Wait until there is sufficient buffer space in |stream| to send the message.
+
+   Note: This means that backpressure will be applied when the user agent's buffers are full.
+
+  1. If [=the WebSocket closing handshake is started|the closing handshake has not yet started=],
+     [=Send a WebSocket Message=] to |stream| comprised of |data| using |opcode|.
+  1. [=Queue a task=] on the [=WebSocket task source=] to resolve |promise| with undefined.
+ 1. Return |promise|.
+</div>
+
+<hr>
+
+<div algorithm="close algorithm">
+
+To <dfn>close</dfn> a {{WebSocketStream}} |stream|, [=close the WebSocket=] with |stream| (<var
+ignore>code</var> and <var ignore>reason</var> are not set).
+</div>
+
+<hr>
+
+<div algorithm="abort algorithm">
+
+To <dfn>abort</dfn> a {{WebSocketStream}} |stream| given |reason|, [=close using reason=] giving
+|stream| and |reason|.
+</div>
+
+<hr>
+
+<div algorithm="close using reason algorithm">
+
+To <dfn>close using reason</dfn> a {{WebSocketStream}} |stream| given |reason|,
+
+ 1. Let |code| and |reasonString| be unset.
```

Comment:
Maybe set these to their default values here, to avoid the "unset".

Reply:
Ah, I missed this step. Fixed.